src/Namespace.sol

```diff
@@ -0,0 +1,578 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.14;
+
+import {ERC721} from "../lib/solmate/src/tokens/ERC721.sol";
+
+import {Owned} from "../lib/solmate/src/auth/Owned.sol";
+
+import {FixedPointMathLib} from "../lib/solmate/src/utils/FixedPointMathLib.sol";
+
+error InsufficientFunds(); // The transaction does not have enough money to pay for this.
+error Unauthorized(); // The caller is not authorized to perform this action.
+
+error InvalidCommit(); // The commitment hash was not found
+error InvalidName(); // The username had invalid characters
+error InvalidTime(); // Time is too far in the future or past
+error IncorrectOwner(); // The username is not owned by the expected address
+
+error Registered(); // The username is currently registered.
+error NotRegisterable(); // The username has been registered and cannot be registered again.
+error Registerable(); // The username has never been registered.
+
+error Expired(); // The username is expired (renewable or biddable)
```

Comment:
Nit: names of errors don't specify that it's for usernames. `ExpiredUsername` would be more explicit